### ISSUE TITLE
[Feature] Improvement of Light Sources Preferences: Appearance, Multiple Lights in the Scene, Background Illumination

### ISSUE THREAD

**Issue by maxwxyz** [FreeCAD/FreeCAD#11166]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

To extend the functionality of https://github.com/FreeCAD/FreeCAD/issues/11113 implemented by https://github.com/FreeCAD/FreeCAD/pull/11146 it would be great to add, adjust, and remove lights in the preference dialog of FreeCAD.
The arrows could indicate a light each.
![grafik](https://github.com/FreeCAD/FreeCAD/assets/6246609/c1f166cf-a88b-4185-843d-e740f99e6325)

The direction of the lights should be editable as well as the intensity. If possible, even the color could be adjusted and all saved in the preferences. The current light which is edited could be highlighted in the view, maybe even represent color and intensity on the ball (like a real time preview). It should be possible to add new lights as well as to remove them again.
Maybe themes could provide settings of the default lightning.

There should be a setting to set the ambientColor and ambientIntensity with a default white ambient environment light in the scene, so no completely black areas on the model are visible.

If possible default light settings e.g. three spot, default, ... could be provided and chosen via a dropdown menu and a custom entry could be saved.

### Full version info

```shell
0.22
```

### Subproject(s) affected?

Core

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by xtemp09** [comment]
I would add that the easy switch between light settings, like MisterMakerNL showed in [the screenshot](https://github.com/FreeCAD/FreeCAD/pull/11146#issuecomment-1775520510), would be of great value.

**Comment by maxwxyz** [comment]
@MisterMakerNL could you specify what settings or presets this would include?

**Comment by obelisk79** [comment]
No, I don't necessarily see how this would directly impact my theme. I do worry about over-complicating with too many advanced controls.

What is the problem adding multiple lights like this is intended to solve?

**Comment by maxwxyz** [comment]
The background in the preferences is currently all white. Maybe a theme can set it to the 3D view background. Or is this feature able to pull the 3D view background @wwmayer ?

**Comment by luzpaz** [comment]
Bumping to continue the conversation

**Comment by luzpaz** [comment]
@maxwxyz can you explicitly elaborate on why this multiple light feature is useful ? TIA

**Comment by maxwxyz** [comment]
@luzpaz as @xtemp09 mentioned above https://github.com/FreeCAD/FreeCAD/pull/11146#issuecomment-1775520510
It is useful to see the object. One light a shiny objects can make the parallel faces facing the camera plane all seem white (see #11113 :
![277129880-234a6f71-5c54-431a-b1a9-b97ff876c1b4](https://github.com/FreeCAD/FreeCAD/assets/6246609/a1779ab4-0a54-40d4-ae28-bb4f3c296f80)
viewing from top:
![277129903-a3b0c9b8-5352-4793-9d04-29ba1087ca6a](https://github.com/FreeCAD/FreeCAD/assets/6246609/69d4bfd7-c762-4858-bba9-7a599fee85f5)

When sketching or selecting references you do not see differences in face color or selecting color so it gets pretty hard with assemblies or referencing in sketches. The fix #11146 lets turn the default position of the light which fixes this issue, but with only one light present in the scene and no global/background illumination, all faces not directly illuminated by the one light appear black.
Simple model to show the problems of the fix of the first problem:
![image](https://github.com/FreeCAD/FreeCAD/assets/6246609/51eee454-eb4f-4971-b293-d3a9c9dbbd1a)

Constantly adjusting the default main headlight position is not ideal in a workflow. Other CAD applications have a general background illumination or multiple light, where you can even switch to different presets or even an HDRI as background image (for reflections).
If multiple lights is too difficult to add, maybe a slider for a general background light from all directions would be great. It is not great if the one setup light creates white faces and completely black faces in the model which makes it hard to work with.

The presets @MisterMakerNL referenced are also a great addition and maybe even themes can ship their individual light settings, to display the model in a great light and much nicer user interface

![277399109-0b416b17-c6e4-4118-bf59-ee9d1d99f77d](https://github.com/FreeCAD/FreeCAD/assets/6246609/6cf915ed-05ee-4437-84f3-ebb3de07c07c)


Example of multiple light settings in CATIA: http://catiadoc.free.fr/online/basug_C2/basugbt1701.htm

**Comment by maxwxyz** [comment]
In addition, I propose to change the color / material appearance of the current sphere to be able to see the light source and shadow area better.
current look (sphere is small in the dialog with a huge arrow and not much difference between light and shadow on the sphere):
![grafik](https://github.com/FreeCAD/FreeCAD/assets/6246609/cd61b778-ba11-4434-9f99-db7da96f7bcf)

proposed preview (screenshot as implemented in the material editor):
![grafik](https://github.com/FreeCAD/FreeCAD/assets/6246609/0a44a7fe-9649-44da-864b-46adc8022ea2)

```
Values:
Ambient color: #333333
Diffuse color: #cccccc
Emissive color: #000000
Specular color: #cccccc
Shininess: 0.6
Transparency: 0
```

**Comment by maxwxyz** [comment]
Additionally, the ambient color and ambient light intensity should be a preference and an ambient light should be in the 3D scene by default so completely dark parts (shadow of headlight) are not created and everything should be visible. the ambient color should be 0.2 0.2 0.2 as a default.

**Comment by maxwxyz** [comment]
I've noticed, that the 'Reset' button for this preferences panel also does no work.

**Comment by M4xTheMan** [comment]
Could this be closed too or is it still an issue?

**Comment by maxwxyz** [comment]
IMO yes, there is no possibility to add multiple lights, change individual lights, use background illumination or even an HDRI

**Comment by M4xTheMan** [comment]
Alright, thank you!

**Comment by maxwxyz** [comment]
Demo of how this could look with multiple lights added via addon (should be possible with Python I guess?)
Would also fix: https://github.com/FreeCAD/FreeCAD/issues/14965

https://www.youtube.com/watch?v=E75hptWOSZA

![Image](https://github.com/user-attachments/assets/19a36dd6-4f4f-4ebf-aed0-9d26b4c74cbe)

@obelisk79 @kadet1090 FYI, how complex would such an implementation be?

**Comment by kadet1090** [comment]
Given that this can be done in addon - should be pretty easy to do as we already have everything needed in code. Hardest to do would be UI for this.

**Comment by maxwxyz** [comment]
I mean ideally we would just have a three point light setup as default without a UI in the beginning. Then to adjust it use the existing preference page for light sources?

**Comment by luzpaz** [comment]
cc @oficinerobotica just wanted to put this on your radar (since your tutorial a model are being referenced here)

**Comment by M4xTheMan** [comment]
@kadet1090 ping :-)
